Description of the loan dataset:

I chose to do my final project on the Loan Prediction dataset. The goal of this data set is to determine if a loan would get approved or not depending on the listed variables of the person trying to get the loan.

Here is an example of an entry and its variables of the Loan Prediction dataset:

In [12]:
%matplotlib inline

import pandas
import numpy
import matplotlib

data = pandas.read_csv("TrainingSet.csv")

data.head(1)

Loan_ID Gender Married Dependents Education Self_Employed  \
0  LP001002   Male      No          0  Graduate            No   

   ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0             5849                0.0         NaN             360.0   

   Credit_History Property_Area Loan_Status  
0             1.0         Urban           Y

To make things a bit more formatted, here are the variables and a brief description 


**_______________________________________________________________________**

Variable------------Description

Loan_ID ----------- Unique ID

Gender--------------Male/Female

Married-------------(Y/N)

Dependents----------# of dependents

Education-----------Applicant Education (Graduate/ Under Graduate)

Self_Employed-------Self employed (Y/N)

ApplicantIncome-----Applicant income

CoapplicantIncome---Coapplicant income

LoanAmount----------Loan amount in thousands

Loan_Amount_Term----Term of loan

Credit_History------Boolean value(1(yes)/0(no))

Property_Area-------Urban/ Semi Urban/ Rural

Loan_Status---------Loan approved (Y/N)

**________________________________________________________________________**


Because im using the pandas library to sort through this data it provides me functionality to get a good start with where to go with this problem. Now that we know the variables and their description, the next step is to find the amount of cases I am dealing with so I can start figuring out if there are missing values.

In [9]:
data.describe()

ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
count       614.000000         614.000000  592.000000         600.00000   
mean       5403.459283        1621.245798  146.412162         342.00000   
std        6109.041673        2926.248369   85.587325          65.12041   
min         150.000000           0.000000    9.000000          12.00000   
25%        2877.500000           0.000000         NaN               NaN   
50%        3812.500000        1188.500000         NaN               NaN   
75%        5795.000000        2297.250000         NaN               NaN   
max       81000.000000       41667.000000  700.000000         480.00000   

       Credit_History  
count      564.000000  
mean         0.842199  
std          0.364878  
min          0.000000  
25%               NaN  
50%               NaN  
75%               NaN  
max          1.000000

So now we know that there are 614 cases in this dataset which gives me a basis to start figuring out if there are missing casses that I need to fill in so that the analysis is more accurate.

For example, right away I can see that there are 22 missing values from LoanAmount, 14 missing values from Loan_Amount_Term, and 50 missing values from Credit_History. I want to see how many missing variables we have in total:

In [13]:
def missingNum(x):
    return sum(x.isnull())

print "Missing values per column"
print data.apply(missingNum, axis=0)

Missing values per column
Loan_ID               0
Gender               13
Married               3
Dependents           15
Education             0
Self_Employed        32
ApplicantIncome       0
CoapplicantIncome     0
LoanAmount           22
Loan_Amount_Term     14
Credit_History       50
Property_Area         0
Loan_Status           0
dtype: int64
